stereo_image_proc/src/stereo_image_proc/disparity_node.cpp

```diff
@@ -0,0 +1,240 @@
+
+  // Synchronize callbacks
+  if (approx) {
+    approximate_sync_.reset(new ApproximateSync(
+        ApproximatePolicy(queue_size),
+        sub_l_image_, sub_l_info_,
+        sub_r_image_, sub_r_info_));
+    approximate_sync_->registerCallback(
+      std::bind(&DisparityNode::imageCb, this, _1, _2, _3, _4));
+  } else {
+    exact_sync_.reset(new ExactSync(
+        ExactPolicy(queue_size),
+        sub_l_image_, sub_l_info_,
+        sub_r_image_, sub_r_info_));
+    exact_sync_->registerCallback(
+      std::bind(&DisparityNode::imageCb, this, _1, _2, _3, _4));
+  }
+
+  // TODO(jacobperron): Setup equivalent of dynamic reconfigure with ROS 2 parameters
+  //                    See configCb
+
+  // TODO(jacobperron): Monitoring subscriber status is currently not possible in ROS 2
+  // std::lock_guard<std::mutex> lock(connect_mutex_);
+
+  // Monitor whether anyone is subscribed to the output
+  // ros::SubscriberStatusCallback connect_cb = std::bind(&DisparityNode::connectCb, this);
+
+  pub_disparity_ = create_publisher<stereo_msgs::msg::DisparityImage>("disparity", 1);
+
+  // TODO(jacobperron): Remove when we can be notified of subscriber status
+  connectCb();
+}
+
+// Handles (un)subscribing when clients (un)subscribe
+void DisparityNode::connectCb()
+{
+  // TODO(jacobperron): Uncomment when we can be notified of subscriber status
+  // std::lock_guard<std::mutex> lock(connect_mutex_);
+  // if (pub_disparity_.getNumSubscribers() == 0)
+  // {
+  //   sub_l_image_.unsubscribe();
+  //   sub_l_info_ .unsubscribe();
+  //   sub_r_image_.unsubscribe();
+  //   sub_r_info_ .unsubscribe();
+  // }
+  // else if (!sub_l_image_.getSubscriber())
+  if (!sub_l_image_.getSubscriber()) {
+    image_transport::TransportHints hints(this, "raw");
+    sub_l_image_.subscribe(this, "left/image_rect", hints.getTransport());
+    sub_l_info_.subscribe(this, "left/camera_info");
+    sub_r_image_.subscribe(this, "right/image_rect", hints.getTransport());
+    sub_r_info_.subscribe(this, "right/camera_info");
+  }
+}
+
+void DisparityNode::imageCb(
+  const sensor_msgs::msg::Image::ConstSharedPtr & l_image_msg,
+  const sensor_msgs::msg::CameraInfo::ConstSharedPtr & l_info_msg,
+  const sensor_msgs::msg::Image::ConstSharedPtr & r_image_msg,
+  const sensor_msgs::msg::CameraInfo::ConstSharedPtr & r_info_msg)
+{
+  // Update the camera model
+  model_.fromCameraInfo(l_info_msg, r_info_msg);
+
+  // Allocate new disparity image message
+  auto disp_msg = std::make_shared<stereo_msgs::msg::DisparityImage>();
+  disp_msg->header = l_info_msg->header;
+  disp_msg->image.header = l_info_msg->header;
+
+  // Compute window of (potentially) valid disparities
+  int border = block_matcher_.getCorrelationWindowSize() / 2;
+  int left = block_matcher_.getDisparityRange() + block_matcher_.getMinDisparity() + border - 1;
+  int wtf;
+  if (block_matcher_.getMinDisparity() >= 0) {
+    wtf = border + block_matcher_.getMinDisparity();
```

Comment:
What does `wtf` represent?  I think that could be named clearer

Reply:
I wish I knew, haha. I didn't touch the logic that was here since ROS 1. Any suggestions are appreciated.

Reply:
I... don't even know what that is doing. I'm going to just back away slowly before I go too far down a rabbithole